week-02/assessment/gomoku/src/main/java/learn/gomoku/ui/GameController.java

```diff
@@ -0,0 +1,158 @@
+import learn.gomoku.game.Stone;
+import learn.gomoku.game.Result;
+import learn.gomoku.players.HumanPlayer;
+import learn.gomoku.players.Player;
+import learn.gomoku.players.RandomPlayer;
+
+import javax.xml.namespace.QName;
+import java.util.Scanner;
+
+import static java.lang.Integer.*;
+
+//gc
+public class GameController {
+    public Scanner console = new Scanner(System.in);
+    public char[][] board = new char[15][15];
+    public Gomoku game;
+
+    // Method: readRequiredString
+    private String readRequiredString(String message) {
+        String input = "";
+        while (input.isBlank()) {
+            System.out.println(message);
+            input = console.next();
+        }
+        return input;
+    }
+
+    private int readInt(String message, int min, int max) {
+        int input = parseInt(readRequiredString(message));
+        if (input > max || input < min) {
+            readInt(message, min, max);
+        }
+        return input;
+    }
+
+    private Player getPlayer(int playerNumber) {
+        System.out.println("Player " + playerNumber + " do you want to play as HumanPlayer or RandomPlayer? ");
+        String playerIdentity = console.next();
+        Player player1 = null;
+        if (playerIdentity.equalsIgnoreCase("HumanPlayer")) {
+            System.out.println("What is your name?");
+            String name = console.next();
+            player1 = new HumanPlayer(name);
+        } else if (playerIdentity.equalsIgnoreCase("RandomPlayer")) {
+            player1 = new RandomPlayer();
+        } else {
+            getPlayer(playerNumber);
+        }
+        return player1;
+    }
+
+    private void printBoard(Gomoku game) {
+        char[][] board = new char[15][15];
+
+        for (int i = 0; i < game.getStones().size(); i++) {
+            Stone stone = game.getStones().get(i);
+            if (stone.isBlack()) {
+                board[stone.getRow()][stone.getColumn()] = 'B';
+            } else {
+                board[stone.getRow()][stone.getColumn()] = 'W';
+            }
+        }
+        //numbering column
+        for (int k = 0; k < board[0].length; k++) {
+            if (k != board[0].length - 1) {
+                if (k == 0) {
+                    System.out.print(" " + (k + 1) + " ");
+                } else {
+                    System.out.print((k + 1) + " ");
+                }
+            } else {
+                System.out.println(k + 1);
+            }
+        }
+        //print board
+        //j = row
+        for (int j = 0; j < 15; j++) {
+            //numbering rows
+            System.out.print(j + 1);
+
+            //k = column
+            for (int k = 0; k < 15; k++) {
+
+                if (board[j][k] == 'B' || board[j][k] == 'W') {
+                    System.out.print(board[j][k]);
+                } else {
+                    System.out.print(" -");
+                }
+            }
+            //sep rows
+            System.out.print("\n");
+        }
```

Comment:
This is acceptable. The board is printing properly. It could be nicer and more precise with some use of the string format method (shortcut: souf) but this works.